### PR TITLE
Tries to fix a problem with URLs that started with ?

### PR DESCRIPTION
**Pull request by JJ**
that is perl6/doc#2270 
I really can't follow too well the transformations that are happening inside that, but I managed to contain the problem here, I guess.

### CODE REVIEW

File: lib/Pod/To/HTML.pm [hunk 1]
```diff
@@ -545,17 +545,21 @@ multi sub node2inline(Pod::FormattingCode $node) returns Str {
...
-            if $text ~~ /^'#'/ {
-                # if we have an internal-only link, strip the # from the text.
-                $text = $/.postmatch
-            }
-            $url = url(unescape_html($url));
-            if $url ~~ /^'#'/ {
-                $url = '#' ~ uri_escape( escape_id($/.postmatch) )
-            }
-            return qq[<a href="$url">{$text}</a>]
+                  my $text = node2inline($node.contents);
```
Comment (Altai-man) [inline lib/Pod/To/HTML.pm]: Why indentation increase here?

**Comment by JJ** [comment]
No reason. My editor did it.

**Comment by JJ** [comment]
Can change it back if you want.

**Comment by Altai-man** [comment]
That would be very nice. I mean, it:
* Makes reviewing harder, instead of easier.
* Adds useless clutter in the commit code changes itself, that reader needs to search for what was changed.
* Inconsistency is bad and we should feel bad for increasing it. :)

**Comment by JJ** [comment]
OK, will do. Thanks for the review.

**Comment by Altai-man** [comment]
I'll play with it now a bit... **Please do not merge yet.**

**Comment by JJ** [comment]
Could you check this?

**Comment by Altai-man** [comment]
Sorry, was working. The things I got from this:
* It seems that un-escaped variants of symbols operators are used and somehow routed to by the site. We have links like `&?BLOCK` or `(&)` which are resolved fine regardless of `&` symbol. But it is different for `?`.
* Your fix applies condition that means basically `If string(url) does not start with '?', apply unescape routine to it`. There are two issues:
  * Your proposed regex does not really work, e.t. try out `perl6 -e "say !('&?BLOCK' ~~ /^\?/)"`, it returns True.
  * Even if it'd be working, `unescape` subroutine does not touch `?` symbols - https://github.com/perl6/Pod-To-HTML/blob/master/lib/Pod/To/HTML.pm#L59-L62 does not have `?` character, so your change won't make it work, which I tested out.

Right now I am not really sure what is the policy about escaping or not escaping symbols in URLs, so it may be a wise way to firstly investigate that and see if there is some unified solution we cannot find because of spaghetti.

Second thing: page `routine.html` which has the issue, is created in `htmlify.p6` and using simple print-driven debugging, we can see that it is already escaped correctly... Which is very odd, so there is some other place where it becomes broken(unescaped).

Considering the above, I would rather say this issue is a bit more complex and needs to be investigated more, but I don't see how this PR can fix it.

**Comment by JJ** [comment]
El mié., 22 ago. 2018 a las 16:41, Altai-man (<notifications@github.com>)
escribió:

> Sorry, was working. The things I got from this:
>
>    - It seems that un-escaped variants of symbols operators are used and
>    somehow routed to by the site. We have links like &?BLOCK or (&) which
>    are resolved fine regardless of & symbol. But it is different for ?.
>    - Your fix applies condition that means basically If string(url) does
>    not start with '?', apply unescape routine to it. There are two
>    issues:
>
> Not really. It checks the name of the URL, not the URL.

>
>    -
>       - Your proposed regex does not really work, e.t. try out perl6 -e
>       "say !('&?BLOCK' ~~ /^\?/)", it returns True.
>
>  It's true, it does not start with ?. I don't think it's a problem with
&?BLOCK.

>
>    -
>       - Even if it'd be working, unescape subroutine does not touch ?
>       symbols -
>       https://github.com/perl6/Pod-To-HTML/blob/master/lib/Pod/To/HTML.pm#L59-L62
>       does not have ? character, so your change won't make it work, which
>       I tested out.
>
> It's ' url' the one that changes the symbols, not unescape.


>    -
>
> Right now I am not really sure what is the policy about escaping or not
> escaping symbols in URLs, so it may be a wise way to firstly investigate
> that and see if there is some unified solution we cannot find because of
> spaghetti.
>
> Second thing: page routine.html which has the issue, is created in
> htmlify.p6 and using simple print-driven debugging, we can see that it is
> already escaped correctly... Which is very odd, so there is some other
> place where it becomes broken(unescaped).
>

It's right here, in the code I've "if" ed. You can check if you want. And
it's url who does it, only I don't know where url is set, since it's a
dynamic variable.

Considering the above, I would rather say this issue is a bit more complex
> and needs to be investigated more, but I don't see how this PR can fix it.
>

This PR _does_ fix that issue. Being htmlify.p6 and this module the way it
is, it might cause another...

Cheers

JJ

**Comment by JJ** [comment]
The problem is that this stuff is so convoluted that it's difficult to
test. I seem to remember that I tested it and it worked, but there might be
other stuff involved I don't know about. I am very grateful for your
checking this stuff, I think that it's better to close this and approach
the problem in some other way. Thanks!

**Pull request closed without merging**